crates/bevy_ecs/src/schedule/executor_parallel.rs

```diff
@@ -169,14 +169,17 @@ impl ParallelExecutor {
         systems: &'scope mut [ParallelSystemContainer],
         world: &'scope World,
     ) {
+        #[cfg(test)]
+        let mut started_systems = 0;
```

Comment:
I think that we should remove this test-enabling code unless you have a justification for it (recorded as a comment in the code).

Reply:
This was done to make it compatible with the existing tests. This realistically frontloads the first iteration of the parallel executor, so a lot of this code is just copied from `process_queued_systems`, which affects those tests.

Reply:
Ah got it. Can you leave a comment to this effect?

Reply:
Added.